lib/app/sequence_base.rb

```diff
@@ -686,13 +690,15 @@ def check_resource_against_profile(resource, resource_type, specified_profile =
         end
         if p
           @profiles_encountered << p.url
-          errors = p.validate_resource(resource)
+          resource_validation_errors = Inferno::RESOURCE_VALIDATOR.validate(resource, versioned_resource_class, p.url)
```

Comment:
It looks like `validate_resource` has different side effects than this bit of code,  in that `@test_warnings` gets the warnings in `validate_resource`. As long as that's ok to do in this context, I don't see why not.

Reply:
Paging @Jammjammjamm and/or @radamson for their opinion on whether we're ok to use `validate_resource` here.

Reply:
This whole method (`check_resource_against_profile`) is very suspect.  It is only used in the DSTU2 argo medication sequences, and it has a check at the beginning that requires the resources to be DSTU2 (so this has a very specific use and isn't particularly generic).  I believe that @okeefm was just swapping in the new API for the old one, so he changed the one line to keep the existing behavior (with the new API), right?

`sequence_base` has way too many methods like this that I would like to see cleaned up.  But I'm not sure that's in the scope of this ticket.

Reply:
Using `validate_resource` in this branch of the condition looks like it should be fine to me. It would be nice if `validate_resource` didn't require a profile and we could use it in both branches of the condition, but I don't think that needs to be done in this PR.